.circleci/config.yml

```diff
@@ -0,0 +1,21 @@
+# 
+version: 2
+jobs:
+  build:
+    working_directory: ~/build
+    docker:
+      - image: docker:17.06.0-ce-git
+    steps:
+      - checkout
+      - setup_remote_docker:
+          reusable: true
+          exclusive: true
+      - run:
+          name: Install bash
+          command: apk add --no-cache bash
+      - deploy:
+          name: Build and Push image to Docker Hub
```

Comment:
Ideally is to divide that single `Build & Push` into 2 steps: `build` & `push` where the build happens as "normal" CircleCI step, while `push` happens on `deploy` CircleCI step.

This way it will be more clear what's going on in CircleCI: when the CI just builds an image and when it's actually uploading it to the Hub.